Src/PerformanceCollector/Shared/Implementation/QuickPulse/QuickPulseCollectionStateManager.cs

```diff
@@ -180,6 +233,37 @@ public TimeSpan UpdateState(string instrumentationKey)
             return this.DetermineBackOffs();
         }
 
+        private void UpdateConfiguration(CollectionConfigurationInfo configurationInfo)
+        {
+            // we only get here if Etag in the header is different from the current one, but we still want to check if Etag in the body is also different
+            if (configurationInfo != null && !string.Equals(configurationInfo.ETag, this.currentConfigurationETag, StringComparison.Ordinal))
+            {
+                this.collectionConfigurationErrors.Clear();
+
+                CollectionConfigurationError[] errors = null;
+                try
+                {
+                    errors = this.onUpdatedConfiguration?.Invoke(configurationInfo);
+                }
+                catch (Exception e)
+                {
+                    this.collectionConfigurationErrors.Add(
+                        CollectionConfigurationError.CreateError(
+                            CollectionConfigurationErrorType.CollectionConfigurationFailureToCreateUnexpected,
+                            string.Format(CultureInfo.InvariantCulture, "Unexpected error applying configuration. ETag: {0}", configurationInfo.ETag ?? string.Empty),
+                            e,
+                            Tuple.Create("ETag", configurationInfo.ETag)));
+                }
+
+                if (errors != null)
+                {
+                    this.collectionConfigurationErrors.AddRange(errors);
+                }
+
+                this.currentConfigurationETag = configurationInfo.ETag;
+            }
```

Comment:
If it's null, the old configuration must remain. Null in this context means we failed to get a configuration for some reason, even though there was an indication that it changed. Such situation would indicate a weird networking issue when headers have made it through, but the body of the response could not have been read. We will continue supplying the old configuration's etag to the service in that case, and eventually will update the configuration.